File: web/components/pagination/pagination-demo.ts
/// <reference path="../../../tsd.d.ts" />
import {
  Component, View, FORM_DIRECTIVES, CORE_DIRECTIVES
} from 'angular2/angular2';
import {pagination} from '../../../components/index';

// webpack html imports
let template = require('./pagination-demo.html');

@Component({
  selector: 'pagination-demo'
})
@View({
  template: template,
  directives: [pagination, FORM_DIRECTIVES, CORE_DIRECTIVES]
})
export class PaginationDemo {
  private totalItems:number = 64;
  private currentPage:number = 4;

  private maxSize:number = 5;
  private bigTotalItems:number = 175;
  private bigCurrentPage:number = 1;

  private setPage(pageNo:number):void {
    this.currentPage = pageNo;
  };

  private pageChanged(event:any):void {
    console.log('Page changed to: ' + event.page);
    console.log('Number items per page: ' + event.itemsPerPage);
  };
}

File: web/components/pagination/pagination-demo.dart
import "package:angular2/angular2.dart";
import "package:ng2_strap/pagination/pagination.dart";

@Component(selector: "pagination-demo")
@View(
    templateUrl: "pagination-demo.html",
    directives: const [PAGINATION_DIRECTIVES, FORM_DIRECTIVES, CORE_DIRECTIVES])
class PaginationDemo {
  int totalItems = 64;
  int currentPage = 4;
  int maxSize = 5;
  int bigTotalItems = 175;
  int bigCurrentPage = 1;
  int smallnumPages = 3;
  int numPages = 4;
  void setPage(int pageNo) {
    currentPage = pageNo;
  }

  void pageChanged() {
    print("Page changed to: $currentPage");
  }
}
